test/cli/files.js

```diff
@@ -322,7 +322,11 @@ describe('files', () => runOnAndOff((thing) => {
     return ipfs('cat QmPZ9gcCEpqKTo6aq61g2nXGUhM4iCL3ewB6LDXZCtioEB/dummy')
       .then(() => expect.fail(0, 1, 'Should have thrown an error'))
       .catch((err) => {
+        const message = err.stderr.match(/^Error:(?: Failed to cat file: Error:)? (.*)$/m)[1]
         expect(err).to.exist()
+        expect(message).to.eql(
+          'no link named "dummy" under QmPZ9gcCEpqKTo6aq61g2nXGUhM4iCL3ewB6LDXZCtioEB'
+        )
```

Comment:
This feature is still not complete. Please add tests for multiple levels of nestedness (i.e `/a/b/c/d`) and you will see that after the first level, this PR fails.

Reply:
Thanks, I was more asking if this would be an acceptable location for this logic, knowing that it could possibly have a performance hit when there is a large amount of files on a node.

What should the error message show when a middle level is missing? I am thinking the message should be `no link named "c/d" under a/b` to provide more context to as to what the requested file is.